src/plugins/simulator/media/tag_medium.cpp

```diff
@@ -32,9 +32,9 @@ namespace argos {
          if(strPosIndexMethod == "grid") {
             size_t punGridSize[3];
             if(!NodeAttributeExists(t_tree, "grid_size")) {
-               punGridSize[0] = cArenaSize.GetX();
-               punGridSize[1] = cArenaSize.GetY();
-               punGridSize[2] = cArenaSize.GetZ();
+               punGridSize[0] = static_cast<UInt32>(cArenaSize.GetX());
```

Comment:
Are these casts really necessary? Is it a warning that makes them necessary? Also, this cast is wrong because it loses the sign.

Reply:
Since `punGridSize` is declared as `size_t` the cast should be `static_cast<size_t>(cArenaSize.GetX());`. This is still a conversion to unsigned, but if cArenaSize is negative the simulator should have thrown an exception before we reached this point.